core/src/main/java/org/apache/iceberg/rest/RESTClientFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public interface RESTClientFactory extends Function<Map<String, String>, RESTClient> {
```

Comment:
What is the purpose of this interface? I don't really see how this would be used.

Reply:
Originally, this had more recognizable benefit. I decided to keep it in so that the hierarchy of `REST*` being the interface and then `HTTP*` being the implementing class, like `RESTClient` <-> `HTTPClient`, would be kept.

However, since this is a plain functional interface, I agree it doesn't provide much benefit. Will remove.